plugins/inputs/phpfpm/phpfpm.go

```diff
@@ -277,6 +275,75 @@ func importMetric(r io.Reader, acc telegraf.Accumulator, addr string) (poolStat,
 	return stats, nil
 }
 
+func expandUrls(urls []string) ([]string, error) {
+	addrs := make([]string, 0, len(urls))
+	for _, url := range urls {
+		if isNetworkURL(url) {
+			addrs = append(addrs, url)
+			continue
+		}
+		paths, err := globUnixSocket(url)
+		if err != nil {
+			return nil, err
+		}
+		addrs = append(addrs, paths...)
+	}
+	return addrs, nil
+}
+
+func globUnixSocket(url string) ([]string, error) {
+	pattern, status := unixSocketPaths(url)
+	glob, err := globpath.Compile(pattern)
+	if err != nil {
+		return nil, fmt.Errorf("could not compile glob %q: %v", pattern, err)
+	}
+	paths := glob.Match()
+	if len(paths) == 0 {
+		return nil, nil
+	}
+
+	// globpath.Match() returns the given argument when it's a static path,
+	// i.e., not a glob pattern. In that case, ensure it exists.
+	if len(paths) == 1 && paths[0] == pattern {
+		if _, err := os.Stat(paths[0]); err != nil {
+			if os.IsNotExist(err) {
+				return nil, fmt.Errorf("Socket doesn't exist  '%s': %s", pattern, err)
+			}
+			return nil, err
+		}
+	}
```

Comment:
Since, as you mentioned, Match() will return the path if there are no special glob characters, I think we can remove this too.  We will still display an error when we dial the socket.

Reply:
Done (this required a change to the unit tests due to the error now coming from Dial).